frontend/src/app/modules/global_search/global-search-input.component.ts

```diff
@@ -199,54 +85,78 @@ export class GlobalSearchInputComponent implements OnDestroy {
     event.stopPropagation();
     event.preventDefault();
 
+    // handle click on search button
     if (ContainHelpers.insideOrSelf(this.btn.nativeElement, event.target)) {
       this.submitNonEmptySearch();
     }
   }
 
-  public redirectToWp(id:string) {
-    window.location = this.PathHelperService.workPackagePath(id) as unknown as Location;
+  // load selected item
+  public onChange($event:any) {
+    let selectedOption = $event;
+    if (selectedOption.id) {  // item is a work package element
+      this.redirectToWp(selectedOption.id);
+    } else {                  // item is a 'scope' element
+      this.searchInScope(selectedOption.projectScope);
+    }
   }
 
-  private unregister() {
-    if (this.unregisterGlobalListener) {
-      this.unregisterGlobalListener();
-      this.unregisterGlobalListener = undefined;
+  // load result list for searched term
+  public handleUserInput($event:any) {
+    this.searchTerm = $event;
+    this.globalSearchService.searchTerm = this.searchTerm;
```

Comment:
The old jQuery solution had a debounce that would ensure that not every key entered into the input field will trigger a search. Instead it would wait a bit and see if nothing changes so that we update the GlobalSearchService less often and not send so many requests to the backend. Did you remove that logic intentionally?